logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -107,6 +113,19 @@ class MessageDataSource(
             messageDAO.updateMessageStatus(messageStatus, messageUuid, idMapper.toDaoModel(conversationId))
         }
 
+    override suspend fun updateAssetMessageDownloadStatus(
+        downloadStatus: Message.DownloadStatus,
+        conversationId: ConversationId,
+        messageUuid: String
+    ): Either<CoreFailure, Unit> =
+        wrapStorageRequest {
+            messageDAO.updateAssetDownloadStatus(
+                MapperProvider.assetMapper().fromDownloadStatusToDaoModel(downloadStatus),
```

Comment:
I think also, this mapper it's a good candidate to be moved to the constructor for testing porpoises and consistency